lib/webdriveragent.js

```diff
@@ -287,6 +289,73 @@ class WebDriverAgent {
     }
   }
 
+
+  /**
+   * @typedef {Object} launchWdaViaDeviceCtlOptions
+   * @property {Object} env environment variables for the launching WDA process
+   */
+
+  /**
+   * Launch WDA with preinstalled package with 'xcrun devicectl device process launch'.
+   * The WDA package must be prepared properly like published via
+   * https://github.com/appium/WebDriverAgent/releases
+   * with proper sign for this case.
+   *
+   * When we implement launching XCTest service via appium-ios-device,
+   * this implementation can be replaced with it.
+   *
+   * @param {launchWdaViaDeviceCtlOptions} opts launching WDA with devicectl command options.
+   * @return {Promise<void>}
+   */
+
+  async _launchWdaViaDeviceCtl(opts) {
+    // FIXME: use appium-xcuitest-driver's Devicectl. Maybe it needs to be included in the `this.device`?
```

Comment:
this is actually a great idea to include it into device, we do the same with simctl by including it into simulator intance